addons/web/static/tests/helpers/mock_server.js

```diff
@@ -790,101 +788,174 @@ export class MockServer {
-            let value = "";
-            groupBy.forEach((groupByField) => {
-                value = (value ? value + "," : value) + groupByField + "#";
-                const fieldName = groupByField.split(":")[0];
-                if (fields[fieldName].type === "date") {
-                    value += formatValue(groupByField, record[fieldName]);
+            } else if (type === "datetime") {
+                const date = DateTime.fromSQL(val);
+                if (aggregateFunction === "hour") {
+                    return date.toFormat("HH:00 dd MMM");
+                } else if (aggregateFunction === "day") {
+                    return date.toFormat("yyyy-MM-dd");
+                } else if (aggregateFunction === "week") {
+                    return `W${date.toFormat("WW kkkk")}`;
+                } else if (aggregateFunction === "quarter") {
+                    return `Q${date.toFormat("q yyyy")}`;
+                } else if (aggregateFunction === "year") {
+                    return date.toFormat("yyyy");
                 } else {
-                    value += JSON.stringify(record[groupByField]);
+                    return date.toFormat("MMMM yyyy");
                 }
-            });
-            return value;
+            } else if (Array.isArray(val)) {
+                if (val.length === 0) {
+                    return false;
+                }
+                return type === "many2many" ? val : val[0];
+            } else {
+                return val;
+            }
         }
+
         if (!groupBy.length) {
             const group = { __count: records.length };
             aggregateFields(group, records);
             return [group];
         }
-        const groups = utils.groupBy(records, groupByFunction);
-        let result = Object.values(groups).map((records) => {
-            const res = {
+
+        const groups = {};
+        for (const r of records) {
+            let recordGroupValues = [];
+            for (const gbField of groupBy) {
+                const [fieldName] = gbField.split(":");
+                let value = formatValue(gbField, r[fieldName]);
+                if (!Array.isArray(value)) {
+                    value = [value];
+                }
+                recordGroupValues = value.reduce((acc, val) => {
+                    const newGroup = {};
+                    newGroup[gbField] = val;
+                    if (recordGroupValues.length === 0) {
+                        acc.push(newGroup);
+                    } else {
+                        for (const groupValue of recordGroupValues) {
+                            acc.push({ ...groupValue, ...newGroup });
+                        }
+                    }
+                    return acc;
+                }, []);
+            }
+            for (const groupValue of recordGroupValues) {
+                const valueKey = JSON.stringify(groupValue);
+                groups[valueKey] = (groups[valueKey] || []).concat([r]);
```

Comment:
.push(r)

Reply:
is that what you suggest ?
```suggestion
                groups[valueKey] = groups[valueKey] || [];
                groups[valueKey].push(r);
```

Reply:
for example but on same line is good too

Reply:
I don´t know how to write this with Array.push in one line :-)

Array.push returns the new length of the array
```js
array = (array || []).push(42);
typeof array // = number
```